docs/data/tree-view/rich-tree-view/editing/Validation.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Tooltip from '@mui/material/Tooltip';
+import CheckCircleOutlineIcon from '@mui/icons-material/CheckCircleOutline';
+import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';
+import { RichTreeView } from '@mui/x-tree-view/RichTreeView';
+import { useTreeItem2Utils } from '@mui/x-tree-view/hooks';
+import { TreeItem2, TreeItem2Props } from '@mui/x-tree-view/TreeItem2';
+import { UseTreeItem2LabelInputSlotProps } from '@mui/x-tree-view/useTreeItem2/useTreeItem2.types';
+import { TreeItem2LabelInput } from '@mui/x-tree-view/TreeItem2LabelInput';
+import { MUI_X_PRODUCTS } from './products';
+
+const ERRORS = {
+  REQUIRED: 'The label cannot be empty',
+  INVALID: 'The label cannot contain digits',
+};
+
+interface CustomLabelInputProps extends UseTreeItem2LabelInputSlotProps {
+  error: null | keyof typeof ERRORS;
+}
+
+function CustomLabelInput(props: Omit<CustomLabelInputProps, 'ref'>) {
+  const { error, ...other } = props;
+
+  return (
+    <React.Fragment>
+      <TreeItem2LabelInput {...other} />
+      {error ? (
+        <Tooltip title={ERRORS[error]}>
+          <ErrorOutlineIcon color="error" />
+        </Tooltip>
+      ) : (
+        <Tooltip title="All good!">
+          <CheckCircleOutlineIcon color="success" />
+        </Tooltip>
+      )}
+    </React.Fragment>
+  );
+}
+
+const CustomTreeItem2 = React.forwardRef(function CustomTreeItem2(
+  props: TreeItem2Props,
+  ref: React.Ref<HTMLLIElement>,
+) {
+  const [error, setError] = React.useState<null | keyof typeof ERRORS>(null);
+  const {
+    interactions: { handleCancelItemLabelEditing, handleSaveItemLabel },
```

Comment:
```suggestion
    interactions,
```

Nitpick, that way we know which methods are the ones returned by this hook when called.
Same for other demos if any